### PR TITLE
Bring back the option to use libnice

### PR DESCRIPTION
**Pull request by lodoyun**
**Description**

This PR brings back the option to use libnice as an alternative to Nicer. This PR is still a work in progress that will allow us to test some edge cases with both libraries to compare performance.

[] It needs and includes Unit Tests

**Changes in Client or Server public APIs**

[] It includes documentation for these changes in `/doc`.

### CODE REVIEW

**Review by jcague — APPROVED** [review]
LGTM!

File: erizo/src/erizo/DtlsTransport.cpp [hunk 1]
```diff
@@ -106,7 +107,11 @@ DtlsTransport::DtlsTransport(MediaType med, const std::string &transport_name, c
     iceConfig_.ice_components = comps;
     iceConfig_.username = username;
     iceConfig_.password = password;
-    ice_ = NicerConnection::create(io_worker_, iceConfig_);
+    if (iceConfig_.use_nicer) {
+      ice_ = NicerConnection::create(io_worker_, iceConfig_);
```
[the remark this answers is not in the document]
Comment (lodoyun) [reply erizo/src/erizo/DtlsTransport.cpp:111]: Good point. I'm actually thinking of changing them so both use `=` instead of reset. The point is that I prefer the `create` method to return a shared_ptr instead of a raw pointer.

File: erizo/src/erizo/thread/IOWorker.h [hunk 2]
```diff
@@ -8,12 +8,18 @@
 #include <thread>  // NOLINT
 #include <vector>
 
+#include "./logger.h"
+#include "lib/LibNiceInterface.h"
```
[the remark this answers is not in the document]
Comment (lodoyun) [reply erizo/src/erizo/thread/IOWorker.h:12]: We don't actually include glib directly anywhere in the licode code, so it would require some tweaking in the build process. I could include libnice (agent, etc) directly but I prefer this.
